### ISSUE TITLE
Make file renaming dialog resizable

### ISSUE THREAD

**Issue by ryan-carpenter** [JabRef/jabref#11799]
**Is your suggestion for improvement related to a problem? Please describe.**
The text field for renaming files is usually too small to show the whole file name. This makes it hard to edit the file name, because the dialog is not resizable. Screen scaling does not matter for 38x40x2160 at 150% or 200% (default font set to 11 pts).

**Describe the solution you'd like**
Make the file-rename dialog wider and/or resizable to enable easier viewing and editing.

**Additional context**
<img src="https://github.com/user-attachments/assets/3d3aec62-194e-42e6-90ed-9fb4b6eee198" width="200" height="">

**Comment by Niharika0612** [comment]
I would like to pick this up.

**Comment by Niharika0612** [comment]
hi, I am getting this issue
org.jabref.logic.citationstyle.CitationStyleTest > numericPropertyMatches(boolean, String) [1] true, ieee.csl FAILED

  org.opentest4j.AssertionFailedError: Citation style not found: ieee.csl ==> expected: <true> but was: <false>

org.jabref.logic.citationstyle.CitationStyleTest > numericPropertyMatches(boolean, String) [2] false, apa.csl FAILED

  org.opentest4j.AssertionFailedError: Citation style not found: apa.csl ==> expected: <true> but was: <false>

org.jabref.logic.citationstyle.CitationStyleTest > numericPropertyMatches(boolean, String) [3] true, vancouver.csl FAILED

  org.opentest4j.AssertionFailedError: Citation style not found: vancouver.csl ==> expected: <true> but was: <false>

org.jabref.logic.citationstyle.CitationStyleTest > numericPropertyMatches(boolean, String) [4] false, chicago-author-date.csl FAILED

  org.opentest4j.AssertionFailedError: Citation style not found: chicago-author-date.csl ==> expected: <true> but was: <false>

org.jabref.logic.citationstyle.CitationStyleTest > numericPropertyMatches(boolean, String) [5] true, nature.csl FAILED

  org.opentest4j.AssertionFailedError: Citation style not found: nature.csl ==> expected: <true> but was: <false>


When i am trying to run the project on my local system.

**Comment by Niharika0612** [comment]
i have checked already and all the files are updates with the main repo

**Comment by koppor** [comment]
@Niharika0612 please try https://devdocs.jabref.org/code-howtos/faq.html#orgjabreflogiccitationstylecitationstyle-discovercitationstyles-error-could-not-find-any-citation-style-tried-with-ieeecsl

**Comment by Niharika0612** [comment]
I tried, and it said to execute these commands, which i did
![git](https://github.com/user-attachments/assets/51cde6c9-bebb-4f0f-8e32-a70a81a569e0)

But my csl-styles and the csl-locales files come up empty. i checked the past history, but i have not made any changes in the file. How do i get the files in this case?

**Comment by Siedlerchr** [comment]
Go int the "jabref" directory, open a terminal and run "git submodule init" and then "git submodule update"

**Comment by Niharika0612** [comment]
![image](https://github.com/user-attachments/assets/3a612f77-a1dd-41ab-b448-fdd8a6283000)
the csl-styles folder still doesn't work

**Comment by ThiloteE** [comment]
Does the following help? `git submodule deinit -f .` then `git submodule update --init --recursive`.

**Comment by Siedlerchr** [comment]
that looks like you had some connection issuses

**Comment by subhramit** [comment]
> But my csl-styles and the csl-locales files come up empty. i checked the past history, but i have not made any changes in the file. How do i get the files in this case?

Hi @Niharika0612. Did you follow the [steps to set up a local development environment](https://devdocs.jabref.org/getting-into-the-code/guidelines-for-setting-up-a-local-workspace/pre-03-code.html)? Did you use the `--recurse-submodules` flag when cloning the repository, as mentioned?

**Comment by Niharika0612** [comment]
> > But my csl-styles and the csl-locales files come up empty. i checked the past history, but i have not made any changes in the file. How do i get the files in this case?
> 
> Hi @Niharika0612. Did you follow the [steps to set up a local development environment](https://devdocs.jabref.org/getting-into-the-code/guidelines-for-setting-up-a-local-workspace/pre-03-code.html)? Did you use the `--recurse-submodules` flag when cloning the repository, as mentioned?

I missed this. I executed this again, so now all the files are here.

**Comment by Niharika0612** [comment]
![image](https://github.com/user-attachments/assets/efec3992-d879-4cb7-b87e-ee79540a2921)
How do i resolve this error? I updates my antler version, but it still does not work.

**Comment by subhramit** [comment]
> How do i resolve this error? I updates my antler version, but it still does not work.

Check out the developer FAQs:

https://devdocs.jabref.org/code-howtos/faq.html#bstvmvisitorjava157-error-package-bstparser-does-not-exist

**Comment by Niharika0612** [comment]
Anyone has any ideas?

**Comment by Siedlerchr** [comment]
Did you run ./gradlew clean and ./gradlew assemble ?

**Comment by Niharika0612** [comment]
> Did you run gralde clean and gradle assemble

Yes I did, but the error still remains. Any idea where the BstBaseVisitor class is supposed to be? 
i thought antlr will handle that class, but no such thing happened.

**Comment by subhramit** [comment]
> Yes I did, but the error still remains. Any idea where the BstBaseVisitor class is supposed to be? i thought antlr will handle that class, but no such thing happened.

I suggest that you do a clean set-up from scratch. Follow each and every step carefully mentioned in the devdocs for setting up. Ideally, if you do so, you should be able to get it up and running.

You can also ask questions on the [gitter chat](https://app.gitter.im/#/room/#JabRef_jabref:gitter.im).

**Comment by Niharika0612** [comment]
> > > How do i resolve this error? I updates my antler version, but it still does not work.
> > 
> > 
> > Check out the developer FAQs:
> > https://devdocs.jabref.org/code-howtos/faq.html#bstvmvisitorjava157-error-package-bstparser-does-not-exist
> 
> the **BstBaseVisitor** (which is supposed to be derived from the antlr) is also missing from the **Bst.g4** ( which is inside the antlr ) and there is no provision in the FAQs for this. if anyone knows how do i get/derive/find the BstBaseVisitor class ?

yeah, i think so too, though i was thinking if there was a chance to do so, without starting from scratch

**Comment by Niharika0612** [comment]
How do i recreate this error in my system?

**Comment by koppor** [comment]
> How do i recreate this error in my system?

Do you mean how to trigger the described issue?

Have a .bib file. Have a PDF atached to an entry. Right click. Select "Rename ..."

![image](https://github.com/user-attachments/assets/9cef0696-b328-4887-92e6-ed9da68676bd)

The resulting dialog is not resizable:

![image](https://github.com/user-attachments/assets/3337cda8-4d5a-490c-b888-e94534c63c74)

Note that you can search for an arbitrary string in the code using <kbd>Ctrl</kbd>+<kbd>Shift</kbd>+<kbd>F</kbd>. Maybe that helps to locate some code.

Background documentation: https://docs.jabref.org/finding-sorting-and-cleaning-entries/filelinks

**Comment by Niharika0612** [comment]
![image](https://github.com/user-attachments/assets/54a27314-34f5-4f5c-b66e-c70b8f2f9f1f)
This is what my application looks like right now. What do I do after this?

**Comment by subhramit** [comment]
> ![image](https://github.com/user-attachments/assets/54a27314-34f5-4f5c-b66e-c70b8f2f9f1f)
> This is what my application looks like right now. What do I do after this?

Go to the entry editor, right click and rename the pdf name

**Comment by Niharika0612** [comment]
![image](https://github.com/user-attachments/assets/27bec923-a6cf-4589-a639-c04fda3b1fdd)
I have opened the entry editor, but cannot see any rename options here.

**Comment by koppor** [comment]
> Go to the entry editor,

Open it by a double click on the entry.

**Comment by juliusalberto** [comment]
Is this issue still being worked on? If not I would like to be assigned for this issue

**Comment by Niharika0612** [comment]
> Is this issue still being worked on? If not I would like to be assigned for this issue

Yes it is being worked upon by me. Thanks.

**Comment by Satyamkumarnavneet** [comment]
Hey @koppor, I am not able to find the code snippet to fix this issue. Could you please help me?

**Comment by Satyamkumarnavneet** [comment]
Hi @koppor, I hope you're doing well. I've been reviewing the code, but I'm having trouble locating the specific section. Could you please provide some guidance or share any relevant documentation that might help me better understand the codebase? Additionally, I would appreciate it if you could assign me this issue. Thank you for your assistance!

**Comment by koppor** [comment]
Its some JavaFX. 

Note that you can search for an arbitrary string in the code using <kbd>Ctrl</kbd>+<kbd>Shift</kbd>+<kbd>F</kbd>. Maybe that helps to locate some code.

**Comment by Satyamkumarnavneet** [comment]
Hi @koppor,

I’ve addressed the issue you raised. The file renaming dialog has been updated as follows:

- Larger Dialog: The dialog is now wide enough to accommodate long file names, making renaming much easier.
- Resizable Dialog: Added setResizable(true) to allow users to resize the dialog as per their needs, providing flexibility across different screen sizes and resolutions.

You can see the updated functionality in the attached screenshot. Please assign this issue to me so I can open a pull request if these changes look good to you.

<img width="1470" alt="Image" src="https://github.com/user-attachments/assets/6067a5be-b133-485e-8e48-d8e79aa10089" />

**Comment by koppor** [comment]
> Hi @koppor,
> 
> I've resolved the issue, and the file renaming dialog has been updated as follows:
> 
> - Larger Dialog: The dialog is now wide enough to accommodate long file names, making renaming much easier.
> - Resizable Dialog: Added setResizable(true) to allow users to resize the dialog as per their needs, providing flexibility across different screen sizes and resolutions.
> 
> You can see the updated functionality in the attached screenshot. Please assign this issue to me so I can open a pull request if these changes look good to you.
> 
> <img width="1470" alt="Image" src="https://github.com/user-attachments/assets/6067a5be-b133-485e-8e48-d8e79aa10089" />

I don't see any pull request

**Comment by Satyamkumarnavneet** [comment]
> > Hi [@koppor](https://github.com/koppor),
> > I've resolved the issue, and the file renaming dialog has been updated as follows:
> > 
> > * Larger Dialog: The dialog is now wide enough to accommodate long file names, making renaming much easier.
> > * Resizable Dialog: Added setResizable(true) to allow users to resize the dialog as per their needs, providing flexibility across different screen sizes and resolutions.
> > 
> > You can see the updated functionality in the attached screenshot. Please assign this issue to me so I can open a pull request if these changes look good to you.
> > <img alt="Image" width="1470" src="https://private-user-images.githubusercontent.com/76639713/403385152-6067a5be-b133-485e-8e48-d8e79aa10089.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzY5NDYwMzAsIm5iZiI6MTczNjk0NTczMCwicGF0aCI6Ii83NjYzOTcxMy80MDMzODUxNTItNjA2N2E1YmUtYjEzMy00ODVlLThlNDgtZDhlNzlhYTEwMDg5LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNTAxMTUlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjUwMTE1VDEyNTUzMFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTEyMzNhOGY2YzM4N2YwY2E4YWMzNThkNzI0ZTExODk0ZmZlNDQ4Yjc1NGNkMmRiYzc0YTEyYTM4YjJkODYxNTMmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.hdrS0NeR6RreFVyjiMgXqet-xmY8OnxPvpLZHlU7Oxw">
> 
> I don't see any pull request

Apologies for the confusion! I hadn’t submitted the pull request earlier. I’ve now created the pull request with the changes implemented. You can check it [here](https://github.com/JabRef/jabref/issues/11799#).

**Comment by subhramit** [comment]
Hey @Satyamkumarnavneet , welcome to JabRef and congratulations on your first contribution! 
> * Larger Dialog: The dialog is now wide enough to accommodate long file names, making renaming much easier.
> * Resizable Dialog: Added setResizable(true) to allow users to resize the dialog as per their needs, providing flexibility across different screen sizes and resolutions.

I'd like to point out [certain guidelines](https://github.com/JabRef/jabref/blob/main/CONTRIBUTING.md#notes-on-ai-usage) that we like to follow here so that you are aware in your journey ahead.

We will provide feedback on your work soon.

**Comment by Satyamkumarnavneet** [comment]
Hi @subhramit, thank you for the warm welcome and for highlighting my first contribution! I look forward to your feedback and appreciate the shared guidelines to help me align better with the project.

**Comment by koppor** [comment]
@Satyamkumarnavneet The PR https://github.com/JabRef/jabref/pull/12390 was closed automatically, because you deleted the branch. - Do you intend to work on this?

Maybe it is as easy as

    inputDialog.setResizable(true);

Don't forget to add a CHANGELOG.md entry.

If you do not want to continue on this, it is also fine.

**Comment by Satyamkumarnavneet** [comment]
> [@Satyamkumarnavneet](https://github.com/Satyamkumarnavneet) The PR [#12390](https://github.com/JabRef/jabref/pull/12390) was closed automatically, because you deleted the branch. - Do you intend to work on this?
> 
> Maybe it is as easy as
> 
> ```
> inputDialog.setResizable(true);
> ```
> 
> Don't forget to add a CHANGELOG.md entry.
> 
> If you do not want to continue on this, it is also fine.

The changes are complete. Please review them at your convenience.
